src/core/drive/form_submission.ts

```diff
@@ -113,7 +113,7 @@ export class FormSubmission {
     const { initialized, requesting } = FormSubmissionState
 
     if (this.needsConfirmation) {
-      const answer = FormSubmission.confirmMethod(this.confirmationMessage!, this.formElement)
+      const answer = await FormSubmission.confirmMethod(this.confirmationMessage!, this.formElement)
```

Comment:
Switching to an assumption that `FormSubmission.confirmMethod` is a promise, what happens when a user-provided Promise set by `Turbo.setConfirmMethod` throws an exception?

Before the `await`, the `async start()` (the method that wraps these lines) would have to deal with the exception.

After adding the `await`, is that exception swallowed?

Reply:
Throwing an error in the promise raises the `Uncaught (in promise)` error in the `start()` method of FormSubmission: https://github.com/hotwired/turbo/blob/201f8b3260bfbc15635d24c9961ff027747fb046/src/core/drive/form_submission.ts#L122

It doesn't swallow the exception.

Reply:
That's good. In that case, should the default function also be translated into a Promise?

https://github.com/hotwired/turbo/blob/201f8b3260bfbc15635d24c9961ff027747fb046/src/core/drive/form_submission.ts#L53-L55

```typescript
static confirmMethod(message: string, element: HTMLFormElement): Promise<boolean> {
  return Promise.resolve(confirm(message))
} 
```

Reply:
It didn't affect the default method, but we could.

We would also want to change this to expect a Promise too?
```ts
export function setConfirmMethod(confirmMethod: (message: string, element: HTMLFormElement)=>Promise<boolean>) {
  FormSubmission.confirmMethod = confirmMethod
}
```